src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java

```diff
@@ -282,7 +282,6 @@ protected ValueInstantiator _constructDefaultValueInstantiator(DeserializationCo
             final boolean isNonStaticInnerClass = beanDesc.isNonStaticInnerClass();
             if (isNonStaticInnerClass) {
                 // TODO: look for `@JsonCreator` annotated ones, throw explicit exception?
-                ;
```

Comment:
Similarly, to emphasic empty block, I sometimes do use semicolons... to indicate intent, not accidentally leaving block empty.

Reply:
I totally get the idea, which is why i suggested 01f7002 in a different occurence. However, IMHO the empty block itself looks like a code smell to me at first glance (merely b/c my IDE highlights it i guess, but still…).

Avoiding the need for such a placeholder, i'd would have suggested refactoring to something like

```java
            if (!isNonStaticInnerClass) {
                // 18-Sep-2020, tatu: Although by default implicit introspection is allowed, 2.12
                //   has settings to prevent that either generally, or at least for JDK types
                final boolean findImplicit = ctorDetector.shouldIntrospectorImplicitConstructors(beanDesc.getBeanClass());
                _addExplicitConstructorCreators(ctxt, ccState, findImplicit);
                if (ccState.hasImplicitConstructorCandidates()
                        // 05-Dec-2020, tatu: [databind#2962] explicit annotation of
                        //   a factory should not block implicit constructor, for backwards
                        //   compatibility (minor regression in 2.12.0)
                        //&& !ccState.hasExplicitFactories()
                        //  ... explicit constructor should prevent, however
                        && !ccState.hasExplicitConstructors()) {
                    _addImplicitConstructorCreators(ctxt, ccState, ccState.implicitConstructorCandidates());
                }
            } // TODO: … else look for `@JsonCreator` annotated ones, throw explicit exception?
```

But i totally see the inner battle between conciseness vs. expressiveness here.

Reply:
I think that it is reasonable to prefer comment within empty block, come to think of it.
And in some cases just inverting the check (did that in one place now).

There are definitely some remaining idiosyncrasies in the source, based on my preferences. Good to re-validate those over time, reduce distance from "common practices".